src/main/java/io/github/nucleuspowered/nucleus/modules/core/commands/MigrateCommand.java

```diff
@@ -0,0 +1,232 @@
+            if (muteHandler != null) {
+                for (UUID uuid : EssentialCmds.muteList) {
+                    Optional<User> user = Sponge.getServiceManager().provideUnchecked(UserStorageService.class).get(uuid);
+
+                    if (user.isPresent()) {
+                        MuteData data = new MuteData(uuid, Util.getMessageWithFormat("command.mute.defaultreason"));
+                        muteHandler.mutePlayer(user.get(), data);
+                    }
+                }
+
+                src.sendMessage(Util.getTextMessageWithFormat("command.nucleus.migrate.mutes"));
+            }
+
+            // Nicknames
+            Configurable mainConfig = Config.getConfig();
+            CommentedConfigurationNode node = Configs.getConfig(mainConfig).getNode("nick");
+
+            for (Object uuid : node.getChildrenMap().keySet()) {
+                String uniqueId = String.valueOf(uuid);
+                String nick = node.getNode(uniqueId).getString();
+
+                try {
+                    NucleusUser uc = userConfigLoader.getUser(UUID.fromString(uniqueId));
+                    uc.setNickname(nick);
+                } catch (IOException | ObjectMappingException e) {
+                    e.printStackTrace();
+                }
+            }
+
+            src.sendMessage(Util.getTextMessageWithFormat("command.nucleus.migrate.nicks"));
+
+            // Mail
+            if (mailHandler != null) {
+                for (Mail mail : Utils.getMail()) {
+                    Optional<User> playerFrom = Sponge.getServiceManager().provideUnchecked(UserStorageService.class).get(mail.getSenderName());
+                    Optional<User> playerTo = Sponge.getServiceManager().provideUnchecked(UserStorageService.class).get(mail.getRecipientName());
+
+                    if (playerFrom.isPresent() && playerTo.isPresent()) {
+                        mailHandler.sendMail(playerFrom.get(), playerTo.get(), mail.getMessage());
+                    }
+                }
+
+                src.sendMessage(Util.getTextMessageWithFormat("command.nucleus.migrate.mail"));
+            }
+
+            // Locked Weather Worlds
+            for (String world : Utils.getLockedWeatherWorlds()) {
+                UUID uniqueId = UUID.fromString(world);
+                NucleusWorld nw = worldConfigLoader.getWorld(uniqueId);
+                nw.setLockWeather(true);
+            }
+
+            src.sendMessage(Util.getTextMessageWithFormat("command.nucleus.migrate.weather"));
+
+            // Rules
+            RulesConfig rc = rca.getNodeOrDefault();
+            List<String> rules = rc.getRuleSet();
+            rules.addAll(Utils.getRules());
+            rc.setRuleSet(rules);
+            rca.setNode(rc);
+            plugin.saveSystemConfig();
+
+            src.sendMessage(Util.getTextMessageWithFormat("command.nucleus.migrate.rules"));
+
+            src.sendMessage(Util.getTextMessageWithFormat("command.nucleus.migrate.success"));
```

Comment:
Not quite. You say the migration has finished, I think an extra message say when they need to do next is useful - don't want them trying to use Nucleus and EssentialCmds at the same time!